messaging/src/main/java/org/axonframework/eventhandling/replay/ResetMessage.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.replay;
+
+import org.axonframework.messaging.Message;
+
+import java.util.Map;
+
+/**
+ * Represents a Message initiating the reset of an Event Handling Component. A payload of {@code T} can be provided to
+ * support the reset operation handling this message.
+ *
+ * @param <T> the type of payload contained in the message
+ * @author Steven van Beelen
+ * @since 4.4
+ */
+public interface ResetMessage<T> extends Message<T> {
```

Comment:
Did you consider extending `EventMessage`? This approach could resolve a backwards compatibility issue with `@ResetHandler` annotation.

Reply:
Unsure why I didn't do this, as I am pretty sure I had this at the start. Gonna re-evaluate this, thanks for pointing it out.

Reply:
Adjusted, as this imposes a stronger contract and indeed resolves potential backwards compatibility issues for those having the `@ResetHandler` as a meta-annotation.